### DISCUSSION TITLE
ICoreWebView2CompositionController#SendMouseInput does not work when point is not (0, 0)?

### DISCUSSION THREAD

**Discussion by hayashikun** [MicrosoftEdge/WebView2Feedback#4056] [Q&A]
I'm reading and editing the code of [flutter-webview-windows](https://github.com/jnschulze/flutter-webview-windows).

In this library, `Microsoft.Web.WebView2` is installed from [nuget](https://www.nuget.org/packages/Microsoft.Web.WebView2/).
Here, the scroll event is sent to webview by ICoreWebView2CompositionController#SendMouseInput.

https://github.com/jnschulze/flutter-webview-windows/blob/afa494ee749ee5e02ced746eef4e9debad4fd98d/windows/webview.cc#L587

I noticed that scroll works only when the point is (0, 0).
Even if the web page is large enough, it cannot be scrolled if a value other than (0, 0) is given, such as (0, 1) or (0, 6).

SDK: 1.0.1210.39, 1.0.1901.177, and 1.0.1988-prerelease
Runtime: 115.0.1901.203
OS: Win10

**Comment by njuptzwt** [comment]
any updates?

**Comment by bradp0721** [comment]
Is there something the HTML is doing to handle/block the mouse wheel event? Can you try navigating your WebView2 to bing.com and see if the issue still happens?

The [WebView2APISample app](https://github.com/MicrosoftEdge/WebView2Samples/tree/main/SampleApps/WebView2APISample) shows calling [SendMouseInput](https://github.com/MicrosoftEdge/WebView2Samples/blob/main/SampleApps/WebView2APISample/ViewComponent.cpp#L786C27-L786C66) with mouse wheel at a point other than (0, 0). Launch the app, go to Window -> Close WebView, then Window -> WebView Creation Mode -> Visual - DComp, then Window -> Create WebView. That will cause the sample app to use the CoreWebView2CompositionController.

**Comment by njuptzwt** [comment]
it works well in [WebView2APISample app]

**Comment by basyskom-bmeier** [comment]
I am also interested in finding a solution to this problem.

As soon as a coordinate other than (0,0) (for example (1,1)) is passed to `SendMouseInput`, it seems like no respective wheel event is genereated in the webview at all. I added the following event listener via devtools in my webview. With (0,0) I can see output, with (1,1) there is no output. To me it looks like it does not depend on the content of the webview?

```
window.addEventListener("wheel", function(e) { console.log("wheel event: ", e); });
```
